webapp/web/js/domjudge.js

```diff
@@ -403,26 +378,55 @@ function initReload(refreshtime)
 	setTimeout(function() { reloadPage(); }, refreshtime * 1000);
 }
 
-// TODO: make this configurable
-function maybeShowEntryPoint(langid)
+function maybeShowEntryPoint(langid, filename = null)
 {
-	'use strict';
 	var entry_point = document.getElementById('entry_point');
 	var entry_point_text = document.getElementById('entry_point_text');
 	var entry_point_help = document.getElementById('entrypointhelp');
-	var visible = langid == 'java' || langid == 'kt' || langid == 'py2' || langid == 'py3';
-	var display = visible ? 'inline' : 'none';
+
+	var entry_point_desc = getEntryPoints(langid);
+	var display = entry_point_desc ? 'inline' : 'none';
 	entry_point.style.display = entry_point_text.style.display = entry_point_help.style.display = display;
-	if ( visible ) {
-		var filename = langid == 'py2' || langid == 'py3';
-		if ( filename ) {
-			entry_point_text.innerHTML = 'Main file:';
-		} else {
-			entry_point_text.innerHTML = 'Main class:';
+	if ( entry_point_desc ) {
+		entry_point_text.innerHTML = entry_point_desc + ':';
+		entry_point.required = true;
+	} else {
+		entry_point.required = false;
+	}
+
+	if ( filename ) {
+		switch (langid) {
+			case 'java':
+				entry_point.value = entryPointDetectJava(filename);
+				break;
+			case 'kt':
+				entry_point.value = entryPointDetectKt(filename);
+				break;
+			default:
+				entry_point.value = filename;
 		}
 	}
 }
 
+function entryPointDetectJava(filename)
```

Comment:
What about moving this into a per language code snippet in the DB or as a new type of script?

Reply:
if we make this a per language code snippet that we could make the same snippet available via the API as well and use the API endpoint from both the web site and the submit client

Reply:
THen we do have to move it to PHP I guess, but that is possible by using some AJAX of course

Reply:
Moving the detection to API calls both from commandline and web is possible but seems a big enough change vs how it works now that I'd like to move that to a separate feature request.

Reply:
sounds reasonable to do this after the 6.0 release